src/Data/RFC5322.hs

```diff
@@ -124,8 +131,82 @@ special = satisfy isSpecial
 atext :: Parser Word8
 atext = satisfy isAtext
 
+
+-- §3.4 Address Specification
+data Mailbox =
+    Mailbox (Maybe T.Text {- display name -})
+            B.ByteString {- addr spec -}
```

Comment:
I think we need a better type for `addr spec`.  One that distinguishes between
domain literal and dot atom.  So something like:

```haskell
data AddrSpec
  = AddrDotAtom (NonEmpty B.ByteString {-printable ascii-})
  | AddrDomainLiteral B.ByteString
```

See below for further comments.

Reply:
Yeah I see your point... Sure why not...

Reply:
Hm.. on a closer try I'm not sure how I "stitch" it together, since.. what do I do about the local part? Is the type then:

```Haskell
(ByteString {- local part -}, AddrSpec)
```
or should the `AddrSpec` be something like:

```Haskell
data AddrSpec = ByteString {- local part -} AddrSpec
data Domain
  = DomainDotAtom (NonEmpty B.ByteString {-printable ascii-})
  | DomainLiteral B.ByteString
```

The latter seems the better choice to me.

Reply:
Yeah, the idea above was a rough sketch.  Definitely the latter.  Although I think you want
```haskell
data AddrSpec = ByteString {- local part -} Domain
```